extensions/stats/plugin.cc

```diff
@@ -634,29 +647,57 @@ void PluginRootContext::report(::Wasm::Common::RequestInfo& request_info,
   }
 
   // handle server-side (inbound) waypoint proxies specially
-  if (!outbound_ && (metadata_mode_ == MetadataMode::kHostMetadataMode ||
-                     metadata_mode_ == MetadataMode::kClusterMetadataMode)) {
+  if (!outbound_ && metadata_mode_ == MetadataMode::kHostMetadataMode) {
     // in waypoint proxy or ztunnel Server mode, we must remap the "local" node
     // info per request as the proxy is no longer serving a single workload
     auto detached = Wasm::Common::extractEmptyNodeFlatBuffer();
 
     flatbuffers::FlatBufferBuilder fbb;
-    if (metadata_mode_ == MetadataMode::kHostMetadataMode) {
-      if (Wasm::Common::extractPeerMetadataFromUpstreamHostMetadata(fbb)) {
-        detached = fbb.Release();
-      }
-    } else {
-      if (Wasm::Common::extractPeerMetadataFromUpstreamClusterMetadata(fbb)) {
-        detached = fbb.Release();
-      }
+    if (Wasm::Common::extractPeerMetadataFromUpstreamHostMetadata(fbb)) {
+      detached = fbb.Release();
     }
 
     const auto& node =
         *flatbuffers::GetRoot<::Wasm::Common::FlatNode>(detached.data());
     map_node(istio_dimensions_, false, node);
   }
 
-  map(istio_dimensions_, outbound_, peer_node_info.get(), request_info);
+  // handle server-side (inbound) ztunnel proxy specifically
+  if (!outbound_ && metadata_mode_ == MetadataMode::kClusterMetadataMode) {
+    ::Wasm::Common::populateTCPRequestInfo(outbound_, &request_info);
+
+    // Map destination node from peer_node_info
+    map_node(istio_dimensions_, false, peer_node_info.get());
+
+    std::string baggage;
+    if (!getValue({"filter_state", "ambient.source.workloadMetadataBaggage"},
+                  &baggage)) {
+      LOG_ERROR("stats plugin: source baggage filter state not found");
+      return;
+    }
+    std::string workload_labels = baggageToLabels(baggage);
+    if (workload_labels == "") {
+      LOG_ERROR(
+          absl::StrCat("stats plugin: failed to convert baggage to workload "
+                       "labels. Baggage: ",
+                       baggage));
+      return;
+    }
+
+    auto detached = Wasm::Common::extractEmptyNodeFlatBuffer();
+    flatbuffers::FlatBufferBuilder fbb;
+    if (Wasm::Common::extractPeerMetadataFromWorkloadLabels(workload_labels,
```

Comment:
I have https://github.com/istio/proxy/pull/4077/files that should allow direct metadata->fb conversion.

Reply:
This looks great and will simplify the conversion and code.
Tried using the metadata object library in the stats plugin but I'm getting compliation error:

```sh
extensions/stats/plugin.cc:24:10: fatal error: 'extensions/common/metadata_object.h' file not found
#include "extensions/common/metadata_object.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I'm not sure what I'm doing wrong as I [imported the header](https://github.com/ymesika/proxy/blob/temp_workload_object/extensions/stats/plugin.cc#L24) and [updated the BUILD file](https://github.com/ymesika/proxy/blob/temp_workload_object/extensions/stats/BUILD#L46).
Got a tip for fixing it?

Reply:
This is probably from Wasm build version. Need to modify imports in here https://github.com/istio/proxy/blob/master/extensions/BUILD.

Reply:
Thanks. It generates this error:
```sh
ERROR: Traceback (most recent call last):
	File "/home/.cache/bazel/_bazel_user/1e0bb3bee2d09d2e4ad3523530d3b40c/external/v8/bazel/defs.bzl", line 404, column 28, in _v8_target_cpu_transition_impl
		v8_target_cpu = mapping[settings["//command_line_option:cpu"]]
Error: key "wasm" not found in dictionary
ERROR: /home/.cache/bazel/_bazel_user/1e0bb3bee2d09d2e4ad3523530d3b40c/external/v8/BUILD.bazel:3254:10: Errors encountered while applying Starlark transition
ERROR: Analysis of target '//extensions:wasm_image_stats' failed; build aborted:
```